server/pbenchinacan/container-build.sh

```diff
@@ -124,6 +124,9 @@ sed -e "s/localhost/${HOSTNAME_F}/" ${PBINC_SERVER}/lib/config/pbench.httpd.conf
 buildah copy --chown root:root --chmod 0644 $container \
     /tmp/pbench.conf.${$} /etc/httpd/conf.d/pbench.conf
 rm /tmp/pbench.conf.${$}
+# We need to ensure HTTPD listens on port 8080 so container can be optionally
+# run with host networking from a non-root user.
+buildah run $container bash -c "sed -i -e 's/^Listen 80$/Listen 8080/' /etc/httpd/conf/httpd.conf"
```

Comment:
Yes, Apache2 was listening only on port 80 before.  So without a change to make it `Listen` on port `8080` nothing works.  This is the change that actually makes Apache2 work with `8080`.  No change is required to the `VirtualHost` since that does not actually perform any "listens", it just looks at HTTP headers to route appropriately.

That is to say, our `VirtualHost` is not actually performing the intended "virtual host" behavior.  The goal of a `VirtualHost` is so that you can have one IP address handling multiple server host names, allowing you to route HTTP header host/port combinations to different upstream services.  We are not using that functionality.

We could have our KeyCloak instance live along side the Pbench Server proxied by Apache2 with a configuration like this:

```
Listen 8080

<VirtualHost *:8080>
    ServerName pbench-server.in.a.can

    ProxyPreserveHost On
    ProxyPass /api/ http://localhost:8001/api/
    ProxyPassReverse /api/ http://localhost:8001/api/
</VirtualHost>

<VirtualHost *:8080>
    ServerName key-cloak.in.a.can

    ProxyPreserveHost On
    ProxyPass / http://localhost:8090/
    ProxyPassReverse / http://localhost:8090/
</VirtualHost>
```

Where we put into the pods `/etc/hosts` definitions for `pbench-server.in.a.can` and `key-cloak.in.a.can` as aliases of `pbenchinacan` host name provided by the pod creation.

This would let us have one port exposed, but access different services by host name.

We really should remove the use of VirtualHost and just add the proxy and directory statements.

But, I digress.

Reply:
Thanks for the background -- I appreciate it.

Since you've brought it up, in our `server/lib/config/pbench.httpd.conf`, wouldn't we be better off if we omitted the `*:8080`?  Since we have only one IP address and we're listening on only one port, it's just needless complication to explicitly specify the wildcard node and the port, no?

Reply:
I agree, but that is for another PR.

Reply:
This PR changes that exact line (to switch the port from 80 to 8080) -- it could instead simply remove the port and remove the IP wildcard as well.

Why should we have two back-to-back PRs which change the same line of a file??

Reply:
I understand.  But, as an alternative to changing 80 to 8080, we could just remove the port number.  That would be a similarly-sized change to the same line, and it actually reduces the complexity in a couple of ways.

The fact that there are two other open PRs where we _could_ work on this sort of thing doesn't mean that we shouldn't do it here:  once again, this is the line which needs changing now, and I don't see any point to changing it multiple times.  (If we had omitted the port number when we created this definition, then we wouldn't be struggling over editing it now.)

I'm looking forward (I think...) to getting rid of the `VirtualHost` definition (although, I'm not sure that what will replace it will be easier to deal with), but I don't know when that will actually happen, so I feel like we might as well make an improvement here and now.